presto-native-execution/presto_cpp/main/common/Configs.h

```diff
@@ -27,6 +27,13 @@ class ConfigBase {
   /// @param filePath Path to configuration file.
   void initialize(const std::string& filePath);
 
+  /// Adds or replaces value at the given key. Can be used by debugging or
+  /// testing code.
+  /// Returns previous value if there was any.
+  folly::Optional<std::string> setValue(
```

Comment:
Should we add the setValue() to the subclass SystemConfig & NodeConfig, instead of the base class ConfigBase? Similar to the initialize() function. If added to the base class, it means all the existing and new subclasses of ConfigBase will all be mutable if mutable-config is set. However, we haven't implemented the ServerOperation endpoint for NodeConfig yet. Is it your planned design?

Reply:
Having such support in the base class simplifies things for any config based on it.

> If added to the base class, it means all the existing and new subclasses of ConfigBase will all be mutable if mutable-config is set. However, we haven't implemented the ServerOperation endpoint for NodeConfig yet. Is it your planned design?

Yes, if we need that,
In fact I plan something like that for the Query Config too, but that stuff is complicated (check the Task, if interested).
So, we'll see how it will pan out.